### DISCUSSION TITLE
Mock Invoke-AzRestMethod

### DISCUSSION THREAD

**Discussion by jasnicchr** [pester/Pester#2153] [Q&A] [answered]
Hello. I am attempting and failing to Mock the below command. I would really appreciate some guidance. The command is pulling back a list of Azure AD Group Members, in particular, Service Principal members.

`((Invoke-AzRestMethod -Uri "https://graph.microsoft.com/beta/groups/$($Group.id)/members").Content | ConvertFrom-Json).value | Select-Object -Property DisplayName, Id, @{label='OdataType';expression={$_.'@odata.type'}}`

My Mock is as follows:
`Mock -CommandName Invoke-AzRestMethod -MockWith {} -ModuleName MyModuleName -ParameterFilter {$Uri -eq "https://graph.microsoft.com/beta/groups/$($fakeAADGroup.id)/members"}`

The error that I get back is:
`ConvertFrom-Json : Cannot bind argument to parameter "InputObject" because it is null.`

The error makes complete sense, it is expecting some input from the pipeline but I am unsure how I mock this required input data.

**Comment by fflaten** [comment]
Run the real `Invoke-AzRestMethod` against a test-group, save the output, clean/anonymize/simplify it and make your mock return it or something in the same format (`-MockWith { "json here" }`)

**Reply by jasnicchr** [reply]
Thanks for the reply. I have already attempted your suggestion but unfortunately I received the same error message.

**Reply by fflaten** [reply]
I noticed now that I missed the call to `Content`-property in your command. `Invoke-AzRestMethod` returns a object with some http method, http response code + a `Content` property with the json. So you need to wrap your mock-json inside object's `Content`-property. Try: `-MockWith { @{ Content = "your valid json here" } }`

**Comment by nohwnd** [comment, marked as answer]
`Invoke-AzRestMethod` returns an object with the json in the `Content` property, so the mock needs to return the same shape: `-MockWith { @{ Content = 'your valid json here' } }`.
